zerver/tests/test_home.py

```diff
@@ -438,7 +438,7 @@ def test_home_under_2fa_with_otp_device(self) -> None:
     def test_num_queries_for_realm_admin(self) -> None:
         # Verify number of queries for Realm admin isn't much higher than for normal users.
         self.login("iago")
-        with self.assert_database_query_count(51):
+        with self.assert_database_query_count(52):
```

Comment:
We should aim to build this feature without needing to add new database queries to the main page loading experience.

Even if we're doing an `AbstractScheduledJob` subclass rather than having clients responsible for expiry, I think we could have that cross-linked to the `UserStatus table in a way that avoids an additional query.

Reply:
so to reduce the query here, I added an additional field to the `user_status` model -- `status_end_time`, which is either `""` or a string, indicating the expiry of the user status. (this will be useful in showing when the user's status will expire in the modal).

Reply:
Why would it be that rather than a `nullable` `DateTimeField`?

Reply:
ok so the reason for that is, this `status_end_time` field in the `UserStatus` is primarily used for displaying the status end time in the "set user status modal". The modal sends a timestamp to the backend, and I am storing it as it is in UserStatus, to avoid an additional step of converting it back to a timestamp when we send it to the frontend (in case it is stored as `DateTimeField`).

The DateTime Field responsible for clearing out the statuses is rather located at `ScheduledUserStatus`.

Reply:
you're right -- I updated it.